.github/workflows/r_package.yml

```
@@ -159,10 +159,35 @@ jobs:
           $env:TASK = "${{ matrix.task }}"
           conda init powershell
           & "$env:GITHUB_WORKSPACE/.ci/test_windows.ps1"
+  test-r-sanitizers:
+    name: r-package (ubuntu-latest, R-devel, GCC ASAN/UBSAN)
+    timeout-minutes: 60
+    runs-on: ubuntu-latest
+    container:
+      image: rhub/rocker-gcc-san
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v1
+        with:
+          fetch-depth: 5
+          submodules: true
+      - name: install
+        shell: bash
+        run: |
+          Rscript -e "install.packages(c('R6', 'data.table', 'jsonlite', 'testthat'), repos = 'http://cran.r-project.org')"
+          sh build-cran-package.sh
+          Rdevel CMD INSTALL lightgbm_*.tar.gz
```

Do we need `.. || exit -1` here for fast-fail or it is default behavior if we write commands under GitHub Actions `run` directive directly without calling `sh` script?

---

oh yes we should have it. I think that all the contents of `run` are sources as a script (not run line by line), so we do need it.

```suggestion
          sh build-cran-package.sh
          Rdevel CMD INSTALL lightgbm_*.tar.gz || exit -1
```